runtime/gc_include/ObjectAccessBarrierAPI.hpp

```diff
@@ -396,6 +396,54 @@ class MM_ObjectAccessBarrierAPI
 #endif /* defined(J9VM_GC_ALWAYS_CALL_OBJECT_ACCESS_BARRIER) */
 	}
 
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+	VMINLINE UDATA
+	mixedObjectGetFieldSize(J9Class *objectClass)
+	{
+		UDATA size = objectClass->totalInstanceSize;
+		if (0 <= (IDATA)objectClass->lockOffset) {
+			size -= sizeof(j9objectmonitor_t);
+		}
+		return size;
+	}
```

Comment:
how about just expanding mixedObjectGetDataSize with another bool argument 'include/exclude' lockwork that would default to 'include'?